File: hello/hello.c
/*
Kattis hello world program solution in C
*/

#include <stdio.h>
#include <string.h>
#include <assert.h>


void test();
char* answer();
void solve();

int main(int argc, char* argv[]) {
    //cout << argc << argv[1] << endl;
    if (argc > 1 && strncmp(argv[1], "test", 4) == 0)
        test();
    else
        solve();
    return 0;
}

char* answer() {
    // can return address to static array from a function
    static char ans[] = "Hello World\0";
    return ans;
}

void test() {
    assert(strncmp(answer(), "Hello World", 11) == 0);
    printf("All test cases passed...\n");
}

void solve() {
    printf("%s\n", answer());
}

File: hello/python3/hello.py
#!/usr/bin/env python3

# Kattis - hello problem

import sys

def answer():
  a = "Hello World!"
  return a

def solve():
  print(answer())

def test():
  assert answer() == "Hello World!"
  print('all test cases passed...')

if __name__ == "__main__":
  solve()
